lib/bundler/index.rb

```diff
@@ -134,10 +134,16 @@ def size
     def ==(other)
       all? do |spec|
         other_spec = other[spec].first
-        other_spec && (spec.dependencies & other_spec.dependencies).empty? && spec.source == other_spec.source
+        other_spec && dependencies_eql?(spec, other_spec) && spec.source == other_spec.source
       end
     end
 
+    def dependencies_eql?(spec, other_spec)
+      deps       = spec.dependencies.select {|d| d.type != :development }
+      other_deps = other_spec.dependencies.select {|d| d.type != :development }
```

Comment:
Can we say that all index comparisons should disregard development dependencies?  I don't know.  How does it work when you have a gem you're developing, you have dev dependencies in your gemspec, you have `gemspec` in your Gemfile and you run bundle.  It installs dev dependencies now.  Does it use an index to do comparisons?